src/web/client/common/remoteFetchProvider.ts

```diff
@@ -4,103 +4,190 @@
+import * as Constants from './constants';
+import { PORTALS_URI_SCHEME } from './constants';
 import { ERRORS, showErrorDialog } from './errorHandler';
 import { PortalsFS } from './fileSystemProvider';
 import { SaveEntityDetails } from './portalSchemaInterface';
 import { registerSaveProvider } from './remoteSaveProvider';
 import { INFO } from './resources/Info';
 let saveDataMap = new Map<string, SaveEntityDetails>();
 
-export async function fetchData(accessToken: string, entity: string, entityId: string, queryParamsMap: Map<string, string>, entitiesSchemaMap: Map<string, Map<string, string>>, languageIdCodeMap: Map<string, string>, portalFs: PortalsFS, websiteIdToLanguage: Map<string, string>) {
+export async function fetchData(
+    accessToken: string,
+    entity: string,
+    entityId: string,
+    queryParamsMap: Map<string, string>,
+    entitiesSchemaMap: Map<string, Map<string, string>>,
+    languageIdCodeMap: Map<string, string>,
+    portalFs: PortalsFS,
+    websiteIdToLanguage: Map<string, string>
+) {
     let requestUrl = '';
     let requestSentAtTime = new Date().getTime();
     try {
-        let url = '';
-        const dataverseOrgUrl = queryParamsMap.get(ORG_URL) as string;
-        if (entityId) {
-            url = SINGLE_ENTITY_URL_KEY;
-        }
-        else url = MULTI_ENTITY_URL_KEY;
-        requestUrl = getRequestURLForSingleEntity(dataverseOrgUrl, entity, entityId, url, entitiesSchemaMap, 'GET');
+        const dataverseOrgUrl = queryParamsMap.get(Constants.ORG_URL) as string;
+
+        requestUrl = getRequestURL(dataverseOrgUrl, entity, entityId, entitiesSchemaMap, Constants.HttpMethod.GET, false);
         sendAPITelemetry(requestUrl);
+
         requestSentAtTime = new Date().getTime();
         const response = await fetch(requestUrl, {
             headers: getHeader(accessToken),
         });
+
         if (!response.ok) {
             vscode.window.showErrorMessage(ERRORS.BACKEND_ERROR);
             sendAPIFailureTelemetry(requestUrl, new Date().getTime() - requestSentAtTime, response.statusText);
             throw new Error(response.statusText);
         }
+
         sendAPISuccessTelemetry(requestUrl, new Date().getTime() - requestSentAtTime);
-        const data = await response.json();
-        if (data.value?.length >= 0) {
-            for (let counter = 0; counter < data.value.length; counter++) {
-                createContentFiles(data[counter], entity, queryParamsMap, entitiesSchemaMap, languageIdCodeMap, portalFs, dataverseOrgUrl, accessToken, entityId, websiteIdToLanguage);
-            }
-        } else {
-            createContentFiles(data, entity, queryParamsMap, entitiesSchemaMap, languageIdCodeMap, portalFs, dataverseOrgUrl, accessToken, entityId, websiteIdToLanguage);
+
+        const result = await response.json();
+        const data = getParameterizedRequestUrlKey(entity, entityId) === Constants.SINGLE_ENTITY_URL_KEY ? [result] : result.value;
```

Comment:
Something seems odd here. How does this work for 'webfiles'?

Reply:
For single entity query the results are returned as flat list of parameters for that single entity. But in all other cases the entities (or even if its a single object returned as part of filter query) are returned inside "value" parameter as a array of objects. So, for webfiles, and multi-files the array of object for iteration will be result.value. As an improvement to this code we should define the response object in case of multi/webfile and single file fetch here for clarity.